pkg/mock/fileUtils.go

```diff
@@ -37,6 +40,45 @@ func (f *FilesMock) init() {
 	}
 }
 
+func getFileProperties(files map[string]map[string]interface{}, name string) (map[string]interface{}, error) {
+
+	if props, ok := files[name]; ok {
+		return props, nil
+	}
+	return nil, fmt.Errorf("File '%s' not found", name)
+}
+
+func getMode(files map[string]map[string]interface{}, name string) (*os.FileMode, error) {
+
+	props, err := getFileProperties(files, name)
+	if err != nil {
+		return nil, err
+	}
+
+	if c, ok := props[keyMode]; ok {
+		if mode, ok := c.(*os.FileMode); ok {
+			return mode, nil
+		}
+	}
+
+	return nil, fmt.Errorf("Cannot provide mode for file '%s' No file mode has been provided or wrong type", name)
+}
+
+func getContent(files map[string]map[string]interface{}, name string) (*[]byte, error) {
+
+	props, err := getFileProperties(files, name)
+	if err != nil {
+		return nil, err
+	}
+
+	if c, ok := props[keyContent]; ok {
+		if content, ok := c.(*[]byte); ok {
+			return content, nil
+		}
+	}
+	return nil, fmt.Errorf("Cannot provide content for file '%s' No file content has been provided or wrong type", name)
+}
+
```

Comment:
These three methods seem to be only there because you map to interface{} instead of a file-scope struct type.

Reply:
... will change to a struct.